### ISSUE TITLE
Adding GAN to improve the result

### ISSUE THREAD

**Issue by nateanl** [speechLabBcCuny/onssen#11]
Although we don't have the label for testing data during the training process, we can use GAN to generalize the model.

Take PIT for the example, the model architecture is LSTM -> FC. We can regard LSTM as the generator, and the output of LSTM is the generated feature for separation. We can generate features for both training data and testing data. The final goal of G Net is making the features of the training data and testing data to be the same distribution. Then we add another LSTM as the discriminator (D Net). The D Net classifies the feature comes from training data or testing data. 

In other words, we don't need to change the current model, we just need to add another LSTM layer on top of the current LSTM layer, and train the separation part and the D Net alternatively.

**Comment by nateanl** [comment]
Close it now and focus on other plans.
